### ISSUE TITLE
Why some operations are not same in two model files?

### ISSUE THREAD

**Issue by miraclebiu** [syfafterzy/PCB_RPP#7]
In ./reid/models/resnet.py Line164:
1. Why calulating the mean and norm the out2?
```
center = out1.mean(0).unsqueeze(0).expand_as(out1)
out2 = x/x.norm(2,1).unsqueeze(1).expand_as(x)
``` 
out2 is calculated without self.feat and self.feat_bn and the value of center is not used?

2. But in the resnet_rpp.py:
the out2 is :
```
        if self.has_embedding:
            x = self.feat(x)
            x = self.feat_bn(x)
            out2 = x/ x.norm(2,1).unsqueeze(1).expand_as(x)
```
The operations are not same, is there some reason for such implementation?
Is this because it is not used in the model due to the self.FCN=True? So that we can ignore the following operations ?

3. the model add a lot of norm operations, it seems like that when we need an output from the model, we add a norm operation, such as resnet_rpp.py Line 153,159,162, is there any reason?

A little suggestion:
```
>>> import torch
>>> import torch.nn.functional as F
>>> aaa=torch.rand(32,1024,8,4)
>>> feat_F_norm = F.normalize(aaa)
>>> feat_ = aaa/aaa.norm(2,1).unsqueeze(1).expand_as(aaa)
>>> torch.sum(feat_F_norm-feat_)
tensor(0.)
```
you can use F.normalize instead of the operations in the paper, they are same.
